tests/e2e/scale/noobaa/test_hsbench.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import pytest
+from ocs_ci.ocs import hsbench
+from ocs_ci.utility import utils
+from ocs_ci.framework.testlib import (
+    E2ETest, scale, ignore_leftovers
+)
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='function')
+def hsbenchs3(request):
+
+    hsbenchs3 = hsbench.HsBench()
+
+    def teardown():
+        hsbenchs3.cleanup()
+    request.addfinalizer(teardown)
+    return hsbenchs3
+
+
+@scale
+@ignore_leftovers
+@pytest.mark.polarion_id("OCS-2321")
+class TestHsBench(E2ETest):
+    """
+    Test writing one million S3 objects to a single bucket
+    """
+    def test_s3_benchmark_hsbench(self, hsbenchs3):
+        """
+        Test case to test one million objects in a single bucket:
+        * Create RGW user
+        * Create test pod
+        * Install hs S3 benchmark
+        * Run hs S3 benchmark to create 1M objects
+        """
+        # Create RGW user
+        hsbenchs3.create_test_user()
+
+        # Create resource for hsbench
+        hsbenchs3.create_resource_hsbench()
+
+        # Install hsbench
+        hsbenchs3.install_hsbench()
+
+        # Running hsbench
+        hsbenchs3.run_hsbench(num_obj=50000, timeout=1800)
```

Comment:
How the timeout is calculated? did we tried this manually, Is there a way to make this generic, I mean for this many number of objects it will take this much time.

Reply:
This time out is base on manually testing result.  I don't have a way to calculate it yet.  Will look into this.